src/AstSemanticChecker.cpp

```diff
@@ -445,6 +469,30 @@ void AstSemanticChecker::checkAggregator(
         report.addError("Unsupported nested aggregate", inner->getSrcLoc());
     }
 
+    AstClause clauseAggregator;
+    // check whether this aggregate 'aggregator' is mutually dependent with
+    // another aggregate contained in this rule. This is disallowed
+    visitDepthFirst(program, [&](const AstLiteral& literal) {
+        visitDepthFirst(literal, [&](const AstAggregator& agg) {
+            if (agg == aggregator) {
+                // make a dummy clause containing the literal in which the aggregate sits
+                clauseAggregator.addToBody(std::unique_ptr<AstLiteral>(literal.clone()));
+                // compare this dummy aggregate clause with every other aggregate clause
+                // to see if there is a mutual dependence
+                visitDepthFirst(program, [&](const AstLiteral& l) {
```

Comment:
Inspiration for naming is running low here. If you have any names other than literal/l/aggregator/agg/a that reflect their purpose that would be great.